### ISSUE TITLE
[Timeline] Restore locator to Add dialog

### ISSUE THREAD

**Issue by VWoeltjen** [nasa/openmctweb#674]
From @RCarvalhoNASA in [a comment](https://github.com/nasa/openmctweb/issues/659#issuecomment-183090522) on #659:

> At a minimum, the dialog to create a new activity while editing inside a timeline should include the locator option, to make sure I agree I'm putting the new activity inside the currently selected activity/timeline so that I can manipulate the hierarchy as I'd like.

**Comment by VWoeltjen** [comment]
This had previously been the behavior of Add, but was removed with New Edit Mode (think the way this was implemented was not compatible with certain aspects of New Edit Mode.) Important constraint here is that the locator for Add should start with the object of navigation, instead of the root object (e.g. should not be able to Add objects outside of the timeline you are currently editing)

AddAction is now separate from CreateAction, so we could introduce this difference at that level.

**Comment by akhenry** [comment]
Yeah, removing the locator was deliberate (see https://github.com/nasa/openmctweb/issues/491#issuecomment-173316445 and https://github.com/nasa/openmctweb/pull/500#issue-128008908).

I think that the main technical reason was that when you're creating a new timeline, and trying to create activities underneath that, the locator was failing due to new objects not having context capability. The solution was that within Timelines, the selected item (the timeline itself, or a sub activity) would be used as the location in lieu of the locator.

The locator is just not going to work with parents that are nascent (non-persisted) objects, without some non-trivial framework changes.

**Comment by larkin** [comment]
Hrm, I don't think this is a blocker, but I do wonder why the scope of changes to fix is so large.  The locator control should be relatively dumb in terms of how it allows selection.  Seems that it should be configurable and should just "figure it out" without having so much tie in to other functionality.

The separate add action allows us to show a locator (which the create action often skips), and that locator should be easy to configure and shouldn't go too crazy when an object is lacking context.

I suppose the issue here is related to the changes that were made in another ticket to allow the locator to determine a default root when it's initially pointed to a link?

**Comment by akhenry** [comment]
@larkin The problem is not so much how the location allows selection, which I think works ok, but rather how it determines the contextual path of the object being edited.

Yeah, maybe the answer is to refactor the locator to be a bit dumber about establishing default parentage for the object. I'd like to deal with contextualization of new objects though, as it's come up a few times for me during NEM because that capability is implicitly depended on by a number of actions and capabilities.

> I suppose the issue here is related to the changes that were made in another ticket to allow the locator to determine a default root when it's initially pointed to a link?
They're related issues, but there's no causal relationship here I think...The locator was removed because it did not work anyway for new activities inside of new timelines. 

Wasn't an issue in Old Edit Mode because you were always editing a persisted object - This incidentally is the cause of a lot of issues in NEM.

We're going to keep putting out spot fires around this until we really reconsider how editing works.

**Comment by charlesh88** [comment]
Possibly relates to #280
